website/source/docs/providers/aws/d/kms_secrets.html.markdown

```diff
@@ -0,0 +1,67 @@
+---
+layout: "aws"
+page_title: "AWS: aws_kms_secrets"
+sidebar_current: "docs-aws-datasource-kms-secrets"
+description: |-
+	Provides secret data encrypted with the KMS service
+---
+
+# aws\_kms\_secrets
+
+The KMS secrets data source allows you to use data encrypted with the AWS KMS
+service within your resource definitions.
+
+## Note about encrypted data
+
+Using this data provider will allow you to conceal secret data within your
+resource definitions but does not take care of protecting that data in the
+logging output, plan output or state output.
+
+Please take care to secure your secret data outside of resource definitions.
+
+## Example Usage
+
+```
+data "aws_kms_secrets" "db" {
+    secret {
+        name = "master_password"
+        payload = "AQECAHhhE7rnmbnLg..."
+
+        context {
+            foo = "bar"
+        }
+    }
+}
+
+resource "aws_rds_cluster" "rds" {
+    master_username = "root"
+    master_password = "${data.aws_kms_secrets.db.master_password}"
+    ...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `secret` - (Required) One or more encrypted payload definitions from the KMS
+  service.  See the Secret Definitions below.
+
+
+### Secret Definitions
+
+Each secret definition supports the following arguments:
+
+* `name` - (Required) The name to export this secret under in the attributes.
+* `payload` - (Required) Base64 encoded payload, as returned from a KMS encrypt
```

Comment:
I think it might be worth giving an example command line here - my own experience with KMS is that the commands can be tricky to use because of (e.g) double-base-64-encoding and so forth.